src/ModalManager.test.js

```diff
@@ -0,0 +1,495 @@
+      expect(onDone).not.toBeCalled();
+
+      fireEvent.click(getById(testIds.checkinNoteModal).getByText(labelIds.confirmButton));
+
+      expect(onDone).toBeCalled();
+    });
+
+    it('should close "CheckinNoteModal" when onConfirm callback is triggered', () => {
+      const modal = screen.getByTestId(testIds.checkinNoteModal);
+
+      fireEvent.click(getById(testIds.checkinNoteModal).getByText(labelIds.confirmButton));
+
+      expect(modal).not.toBeInTheDocument();
+    });
+  });
+
+  describe('when "checkinNotesMode" is false', () => {
+    describe(`item status is ${statuses.CLAIMED_RETURNED}`, () => {
+      describe('and checkedinItem have numberOfPieces greater than one', () => {
+        const newCheckedinItem = {
+          ...checkedinItem,
+          numberOfPieces: 2,
+        };
+
+        beforeEach(() => {
+          render(
+            <ModalManager
+              {...defaultProps}
+              checkedinItem={newCheckedinItem}
+              checkinNotesMode={false}
+            />
+          );
+        });
+
+        it('should render "ClaimedReturnedModal" with correct props', () => {
+          expect(ClaimedReturnedModal).toHaveBeenCalledWith({
+            item: newCheckedinItem,
+            open: true,
+            onCancel: expect.any(Function),
+            onConfirm: expect.any(Function),
+          }, {});
+        });
+
+        it('should correctly handle cancel of "ClaimedReturnedModal"', () => {
+          expect(onCancel).not.toBeCalled();
+
+          fireEvent.click(getById(testIds.claimedReturnedModal).getByText(labelIds.cancelButton));
+
+          expect(onCancel).toBeCalled();
+        });
+
+        it('should close "ClaimedReturnedModal" when onCancel callback is triggered', () => {
+          const modal = screen.getByTestId(testIds.claimedReturnedModal);
+
+          fireEvent.click(getById(testIds.claimedReturnedModal).getByText(labelIds.cancelButton));
+
+          expect(modal).not.toBeInTheDocument();
+        });
+
+        it('should correctly handle confirm of "ClaimedReturnedModal"', () => {
+          expect(claimedReturnedHandler).not.toBeCalled();
+
+          fireEvent.click(getById(testIds.claimedReturnedModal).getByText(labelIds.confirmButton));
+
+          expect(claimedReturnedHandler).toBeCalled();
+        });
+
+        it('should close "ClaimedReturnedModal" when onConfirm callback is triggered', () => {
+          const modal = screen.getByTestId(testIds.claimedReturnedModal);
+
+          fireEvent.click(getById(testIds.claimedReturnedModal).getByText(labelIds.confirmButton));
+
+          expect(modal).not.toBeInTheDocument();
+        });
+
+        describe('after "ClaimedReturnedModal" confrimation', () => {
+          beforeEach(() => {
+            fireEvent.click(getById(testIds.claimedReturnedModal).getByText(labelIds.confirmButton));
+          });
+
+          afterEach(() => {
+            jest.clearAllMocks();
```

Comment:
we use it twice, I would move it to the first describe

Reply:
Fixed